plugin/plugin.go

```diff
@@ -117,13 +159,59 @@ func (GenerateEnvelopeRequest) Command() Command {
 	return CommandGenerateEnvelope
 }
 
-// GenerateSignatureResponse is the response of a generate-envelope request.
+// GenerateEnvelopeResponse is the response of a generate-envelope request.
 type GenerateEnvelopeResponse struct {
 	SignatureEnvelope     []byte            `json:"signatureEnvelope"`
 	SignatureEnvelopeType string            `json:"signatureEnvelopeType"`
 	Annotations           map[string]string `json:"annotations,omitempty"`
 }
 
+// VerifySignatureRequest contains the parameters passed in a verify-signature request.
+type VerifySignatureRequest struct {
+	ContractVersion string            `json:"contractVersion"`
+	Signature       Signature         `json:"signature"`
+	TrustPolicy     TrustPolicy       `json:"trustPolicy"`
+	PluginConfig    map[string]string `json:"pluginConfig,omitempty"`
+}
+
+// Signature represents a signature pulled from the envelope
+type Signature struct {
+	CriticalAttributes    CriticalAttributes `json:"criticalAttributes"`
+	UnprocessedAttributes []string           `json:"unprocessedAttributes"`
+	CertificateChain      [][]string         `json:"certificateChain"`
```

Comment:
Use `[][]bytes` for consistency. `encoding/json` is smart enough to encode a []byte into a base64 string.